File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
bool func(string s, string t){
        unordered_map<char,char>map;
     
        for(int i=0;i<s.length();i++)
        { 
           if((int)map[s[i]]==0)
           map[s[i]]=t[i];
           else
           {
            if(map[s[i]]!=t[i])
            return false;
           }
        }

        return true;
}
    bool isIsomorphic(string s, string t) {
        
        return func(s,t)&&func(t,s);
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean func(String s, String t){

        Map<Character,Character>map=new HashMap<>();
        for(int i=0;i<s.length();i++){

            if(map.get(s.charAt(i))==null)
            map.put(s.charAt(i),t.charAt(i));
            else{
                if(map.get(s.charAt(i))!=t.charAt(i))
                return false;
            }
        }

        return true;
    }
    public boolean isIsomorphic(String s, String t) {
        return func(s,t)&&func(t,s);
    }
}